main.tf

```diff
@@ -1,18 +1,21 @@
 locals {
-  enabled                    = module.this.enabled ? 1 : 0
-  parent_zone_record_enabled = var.parent_zone_record_enabled && module.this.enabled ? 1 : 0
+  enabled                    = module.this.enabled
+  parent_zone_record_enabled = local.enabled && var.parent_zone_record_enabled
+  public_zone                = try(length(var.private_hosted_zone_vpc_attachments), 0) == 0
```

Comment:
@korenyoni You cannot have `public_zone` be depended on derived data like this, because you need it to be defined at `terrform plan` time. Add a boolean input like `soa_record_enabled` or `is_private_zone` (check with @osterman and @aknysh about which they prefer) and use that.